enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/config/ConfigHolder.java

```diff
@@ -504,13 +515,61 @@ private void populateJWTGeneratorConfigurations(JWTGenerator jwtGenerator) {
         jwtConfigurationDto.setEnableUserClaims(jwtGenerator.getEnableUserClaims());
         jwtConfigurationDto.setGatewayJWTGeneratorImpl(jwtGenerator.getGatewayGeneratorImpl());
         jwtConfigurationDto.setTtl(jwtGenerator.getTokenTtl());
+        List<Keypair> keypairs = jwtGenerator.getKeypairsList();
+        // Validation is done at the adapter to ensure only one signing keypair is sent here.
+        Keypair signingKey = getSigningKey(keypairs);
+
         try {
-            jwtConfigurationDto.setPublicCert(TLSUtils.getCertificate(jwtGenerator.getPublicCertificatePath()));
-            jwtConfigurationDto.setPrivateKey(JWTUtils.getPrivateKey(jwtGenerator.getPrivateKeyPath()));
+            jwtConfigurationDto.setPublicCert(TLSUtils.getCertificate(signingKey.getPublicCertificatePath()));
+            jwtConfigurationDto.setPrivateKey(JWTUtils.getPrivateKey(signingKey.getPrivateKeyPath()));
         } catch (EnforcerException | CertificateException | IOException e) {
             logger.error("Error in loading public cert or private key", e);
         }
         config.setJwtConfigurationDto(jwtConfigurationDto);
+        populateBackendJWKSConfiguration(jwtGenerator);
+    }
+
+    private void populateBackendJWKSConfiguration(JWTGenerator jwtGenerator) {
+        BackendJWKSDto backendJWKSDto = new BackendJWKSDto();
+        List<Keypair> keypairs = jwtGenerator.getKeypairsList();
+        ArrayList<JWK> jwks = new ArrayList<>();
+        try {
+            for (Keypair keypair : keypairs) {
+                X509Certificate cert = X509CertUtils
+                        .parse(TLSUtils.getCertificate(keypair.getPublicCertificatePath()).getEncoded());
+                RSAPublicKey publicKey = RSAKey.parse(cert).toRSAPublicKey();
+                RSAKey jwk = new RSAKey.Builder(publicKey)
+                        .keyUse(KeyUse.SIGNATURE)
+                        .algorithm(getJWKSAlgorithm(jwtGenerator.getSigningAlgorithm()))
+                        .keyIDFromThumbprint()
+                        .build().toPublicJWK();
+                jwks.add(jwk);
+            }
+        } catch (JOSEException | CertificateException | IOException e) {
+            logger.error("Error in loading additional public certs for JWKS", e);
```

Comment:
Let's use proper error logging https://github.com/wso2/product-microgateway/blob/main/developer-guide.md#error-logging-in-enforcer

Add error codes to here https://docs.google.com/spreadsheets/d/1ZH2A96qdXF-4FyLsw2pBuWOO6d7tGKH4FmDYvxrPiBM/edit#gid=2045311140

Reply:
ack